extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java

```diff
@@ -198,6 +213,110 @@ void registerSecurityEventObserver(Capabilities capabilities, OTelBuildConfig bu
         }
     }
 
+    /**
+     * Generates {@link SpanProcessor} that adds end-user attributes to new Spans.
+     *
+     * Generates:
+     *
+     * <code>
+     * import io.quarkus.arc.Unremovable;
+     * import io.quarkus.opentelemetry.runtime.tracing.security.SecurityEventUtil;
+     * import jakarta.enterprise.context.Dependent;
+     *
+     * import io.opentelemetry.context.Context;
+     * import io.opentelemetry.sdk.trace.ReadWriteSpan;
+     * import io.opentelemetry.sdk.trace.ReadableSpan;
+     * import io.opentelemetry.sdk.trace.SpanProcessor;
+     *
+     * &#064;Dependent
+     * &#064;Unremovable
+     * public class EndUserSpanProcessor implements SpanProcessor {
+     *
+     *     &#064;Override
+     *     public void onStart(Context parentContext, ReadWriteSpan span) {
+     *         SecurityEventUtil.addEndUserAttributes(span);
+     *     }
+     *
+     *     &#064;Override
+     *     public boolean isStartRequired() {
+     *         return Boolean.TRUE;
+     *     }
+     *
+     *     &#064;Override
+     *     public void onEnd(ReadableSpan span) {
+     *     }
+     *
+     *     &#064;Override
+     *     public boolean isEndRequired() {
+     *         return Boolean.FALSE;
+     *     }
+     *
+     * }
+     * </code>
+     *
+     */
+    @BuildStep(onlyIf = EndUserAttributesEnabled.class)
+    void generateEndUserAttributesSpanProcessor(BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            Capabilities capabilities) {
+        if (capabilities.isPresent(Capability.SECURITY)) {
+            ClassOutput classOutput = new GeneratedBeanGizmoAdaptor(generatedBeans);
+
+            try (ClassCreator classCreator = ClassCreator.builder().classOutput(classOutput)
+                    .className("io.quarkus.opentelemetry.runtime.tracing.security.EndUserSpanProcessor")
```

Comment:
Couldn't this be done with a synthetic bean instead?
One with the source code as a pojo that then is upgraded to a bean?